plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/catalog/jdbc/TrinoJdbcCatalog.java

```diff
@@ -0,0 +1,348 @@
+        this.jdbcCatalog = requireNonNull(jdbcCatalog, "jdbcCatalog is null");
+        this.fileSystemFactory = requireNonNull(fileSystemFactory, "fileSystemFactory is null");
+        this.defaultWarehouseDir = requireNonNull(defaultWarehouseDir, "defaultWarehouseDir is null");
+    }
+
+    @Override
+    public boolean namespaceExists(ConnectorSession session, String namespace)
+    {
+        return jdbcCatalog.namespaceExists(Namespace.of(namespace));
+    }
+
+    @Override
+    public List<String> listNamespaces(ConnectorSession session)
+    {
+        return jdbcCatalog.listNamespaces().stream()
+                .map(namespace -> namespace.level(0).toLowerCase(ENGLISH))
+                .collect(toImmutableList());
+    }
+
+    @Override
+    public Map<String, Object> loadNamespaceMetadata(ConnectorSession session, String namespace)
+    {
+        return jdbcCatalog.loadNamespaceMetadata(Namespace.of(namespace)).entrySet().stream()
+                .collect(toImmutableMap(Map.Entry::getKey, Map.Entry::getValue));
+    }
+
+    @Override
+    public Optional<TrinoPrincipal> getNamespacePrincipal(ConnectorSession session, String namespace)
+    {
+        return Optional.empty();
+    }
+
+    @Override
+    public void createNamespace(ConnectorSession session, String namespace, Map<String, Object> properties, TrinoPrincipal owner)
+    {
+        try {
+            jdbcCatalog.createNamespace(Namespace.of(namespace), transformValues(properties, String.class::cast));
+        }
+        catch (RuntimeException e) {
+            throw new TrinoException(ICEBERG_CATALOG_ERROR, "Failed to create a namespace " + namespace, e);
+        }
+    }
+
+    @Override
+    public void dropNamespace(ConnectorSession session, String namespace)
+    {
+        jdbcCatalog.dropNamespace(Namespace.of(namespace));
+    }
+
+    @Override
+    public void setNamespacePrincipal(ConnectorSession session, String namespace, TrinoPrincipal principal)
+    {
+        throw new TrinoException(NOT_SUPPORTED, "setNamespacePrincipal is not supported for Iceberg JDBC catalog");
+    }
+
+    @Override
+    public void renameNamespace(ConnectorSession session, String source, String target)
+    {
+        throw new TrinoException(NOT_SUPPORTED, "renameNamespace is not supported for Iceberg JDBC catalog");
+    }
+
+    @Override
+    public List<SchemaTableName> listTables(ConnectorSession session, Optional<String> namespace)
+    {
+        List<String> namespaces = listNamespaces(session, namespace);
+        // Build as a set and convert to list for removing duplicate entries due to case difference
+        ImmutableSet.Builder<SchemaTableName> tablesListBuilder = ImmutableSet.builder();
+        for (String schemaName : namespaces) {
+            jdbcCatalog.listTables(Namespace.of(schemaName)).forEach(table -> tablesListBuilder.add(new SchemaTableName(schemaName, table.name())));
```

Comment:
Catch a `NoSuchNamespaceException` from `jdbcCatalog.listTables` and ignore it. May happen that a schema is dropped between collecting the Namespaces and loading the tables from each.

Reply:
Good catch, updated.